src/commands/src/shared/next-gen-deploy.ts

```diff
@@ -1,38 +1,76 @@
-  }
+/* This is still in development - use with caution */
+/* Logs here for future debugging purposes */
+export async function nextGenDeploy({ mutPrefix, gitBranch, hasConfigRedirects }: NextGenDeployParams) {
+  try {
+    if (hasConfigRedirects && (gitBranch === 'main' || gitBranch === 'master')) {
+      // equivalent to: mut-redirects config/redirects -o public/.htaccess
+      await executeCliCommand({ command: 'mut-redirects', args: ['config/redirects', '-o', 'public/.htaccess'] });
+      console.log(`COMMAND: mut-redirects config/redirects -o public/.htaccess`);
+    }
 
-  // equivalent to: yes | mut-publish public ${BUCKET} --prefix="${MUT_PREFIX}" --deploy --deployed-url-prefix=${URL} --json --all-subdirectories ${ARGS};
-  const { outputText } = await executeAndPipeCommands(
-    { command: 'yes' },
-    {
-      command: 'mut-publish',
-      args: [
-        'public',
-        bucket,
-        `--prefix=${mutPrefix}`,
-        '--deploy',
-        `--deployed-url-prefix=${url}`,
-        '--json',
-        '--all-subdirectories',
-      ],
-      options: {
-        cwd: `${process.cwd()}/snooty`,
-      },
+    const bucket = process.env.BUCKET;
+    const url = process.env.URL;
+    if (!bucket) {
+      console.log(`nextGenDeploy has failed. Variable for S3 bucket address was undefined.`);
+      return {
+        status: 'failure',
+        output: 'Failed in nextGenDeploy: No value present for S3 bucket',
+        error: 'No value present for S3 bucket.',
+      };
+    }
+    if (!url) {
+      console.log(`nextGenDeploy has failed. Variable for URL address was undefined.`);
+      return {
+        status: 'failure',
+        output: 'Failed in nextGenDeploy: No value present for target url.',
+        error: 'No value present for URL.',
+      };
     }
-  );
 
-  return `${outputText}\n Hosted at ${url}/${mutPrefix}`;
+    console.log(`URL: ${url}\nBUCKET: ${bucket}\n`);
+
+    // equivalent to: yes | mut-publish public ${BUCKET} --prefix=${MUT_PREFIX} --deploy --deployed-url-prefix=${URL} --json --all-subdirectories ${ARGS};
+    const { outputText } = await executeAndPipeCommands(
+      { command: 'yes' },
+      {
+        command: 'mut-publish',
+        args: [
+          'public',
+          bucket,
+          `--prefix=${mutPrefix}`,
+          '--deploy',
+          `--deployed-url-prefix=${url}`,
+          '--json',
+          '--all-subdirectories',
+        ],
+        options: {
+          cwd: `${process.cwd()}/snooty`,
+        },
+      }
+    );
+    console.log(
+      `COMMAND: yes | mut-publish public ${bucket} --prefix=${mutPrefix} --deploy --deployed-url-prefix=${url} --json --all-subdirectories ${process.cwd()}/snooty`
+    );
+    console.log(`${outputText}\n Hosted at ${url}/${mutPrefix}`);
+    return {
+      status: 'inProgress',
+      output: outputText,
+      error: '',
+    };
+  } catch (error) {
+    console.log(`nextGenDeploy has failed.`);
+    return {
+      status: 'failure',
```

Comment:
So glad you saw this. I remember having an error (which had nothing to do with this) that had "Invalid Status" and I was sure this just had the wrong value. 

But [here is where](https://github.com/mongodb/docs-worker-pool/blob/c15b72c82c6b7ddab666fbb351132f0011c70dd4/src/services/commandExecutor.ts#L60) the "strings" are defined. We/I should change the typescript from `string` to "success"|"failed".

This is great to realize because some places I put "failed" and some I put "failure". Sloppy. Thank you for seeing.

Reply:
They are also defined as an enum [here](https://github.com/mongodb/docs-worker-pool/blob/01a9d71706d4b0dcc08654278e61b83c53c7180e/src/entities/job.ts#L4) if you want to reuse it. They have string values, so you could do something like `const status: JobStatus = 'failed'` and it should work (I think)

Reply:
Well, these are not JobStatuses sadly, these are... CommandExecutorResponse statuses lol